src/apps/service/apps/jobs/submissions/async.clj

```diff
@@ -0,0 +1,118 @@
+(ns apps.service.apps.jobs.submissions.async
+  (:use [clojure-commons.core :only [remove-nil-values]]
+        [slingshot.slingshot :only [try+ throw+]])
+  (:require [clojure.tools.logging :as log]
+            [clojure-commons.file-utils :as ft]
+            [apps.clients.data-info :as data-info]
+            [apps.clients.notifications :as notifications]
+            [apps.persistence.jobs :as jp]
+            [apps.service.apps.job-listings :as job-listings]
+            [apps.service.apps.jobs.submissions.submit :as submit]
+            [apps.util.config :as config]))
+
+(defn- max-batch-paths-exceeded
+  [max-paths first-list-path first-list-count]
+  (throw+
+    {:type       :clojure-commons.exception/illegal-argument
+     :error      (str "The HT Analysis Path List exceeds the maximum of "
+                      max-paths
+                      " allowed paths.")
+     :path       first-list-path
+     :path-count first-list-count}))
+
+(defn- validate-path-lists
```

Comment:
The old `apps.service.apps.jobs.submissions/validate-path-lists` function used to also check the existence of every path in the HT Path List files, but that's been removed from this namespace since that existence checker endpoint is limited to 1000 paths, and we want to raise the HT Path List limit higher than that.

Reply:
So, just to be sure I understand, what'll happen in case a file in a pathlist doesn't exist is that the job will get submitted with the path, and then porklock will fail to download it, and the subjob will be marked failed?

Reply:
Secondarily, since this is now done async anyway, how bad would it be to just [partition-all](https://clojuredocs.org/clojure.core/partition-all) the list of paths into sets of 1000 or fewer and still check for existence? Letting it fail once it hits condor seems like it'd at least work, but with the relieving of the time pressure by putting this in a separate thread it feels weird to also remove validation like this. But maybe pushing that check out to condor (implicitly) makes the apps service less of a bottleneck? Not sure.

Reply:
Yeah, I'm not sure either. Before we would not submit jobs for missing paths, but I don't think the user ever got a message about the missing paths (the service returned the list of missing paths, but I don't think that was displayed in the UI). So if 1 out of 3 paths was missing, their batch job would successfully complete, but it would only have 2 sub-jobs. I'm not sure that's very UX friendly. At least with the failed job, the user could examine the error logs and see why the job failed (file not found).

If we can find a user-friendly way of letting the user know which paths are missing, then I think we can add back existence checking (whether in batches of 1000 or once per sub-job).

Reply:
That makes sense. I suppose this may just be a better thing to do later; letting the user know which paths are missing seems like it could go in notifications, but it could get too noisy. Maybe the best eventual place for this would be if we eventually add something like a job history view, we could send one notification if any paths are missing, which sends the user to the history (which should have events marked for the missing files).

Anyway, now that you've reminded me of the prior behavior of just not submitting jobs, I do think you're right and this is better.